File: GUIGenerator/src/main/java/GUIGenerator/htmlelements/HTMLElement.java
package SketchBuilderCodeGenerator.htmlelements;

import SketchBuilderCodeGenerator.elements.Element;
import SketchBuilderCodeGenerator.languagewrite.Tag;

public abstract class HTMLElement implements Element {
    public abstract void draw(Tag root);

    private String element;
    private String id;
    private String start_column;
    private String width;
    private String height;



    public HTMLElement ( String element ,String id,String start_column ,  String width , String height ){
        this.element = element;
        this.id = id;
        this.start_column = start_column;
        this.width = width;
        this.height = height;

    }

    public String getElement() {
        return element;
    }
    public void setElement(String element) {
        this.element = element;
    }

    public String getId() {
        return id;
    }
    public void setId(String id) {
        this.id = id;
    }

    public String getStart_column() {
        return start_column;
    }
    public void setStart_column(String start_column) {
        this.start_column = start_column;
    }


    public String getWidth() {
        return width;
    }
    public void setWidth(String top) {
        this.width = width;
    }

    public String getHeight() {
        return height;
    }
    public void setHeight(String top) {
        this.height = height;
    }




}

File: Postprocessor/src/HtmlELement.cpp

#include <string>

#include "equalizer/HtmlElement.hpp"

const std::string HtmlElement::ELEMENT_KEY = "element";
const std::string HtmlElement::ID_KEY = "id";
const std::string HtmlElement::START_COLUMN_KEY = "start_column";
const std::string HtmlElement::TOP_KEY = "top";
const std::string HtmlElement::WIDTH_KEY = "width";
const std::string HtmlElement::HEIGHT_KEY = "height";


HtmlElement::HtmlElement(){}
HtmlElement::HtmlElement(std::string elementType, std::string id, int startColumn, int top, int width, int height){
    this->element = elementType;
    this->id = id;
    this->start_column = startColumn;
    this->top = top;
    this->width = width;
    this->height = height;
}
HtmlElement::~HtmlElement(){}
    

std::string HtmlElement::getElementType()const{return this->element;}
std::string HtmlElement::getId()const{return this->id;}
int HtmlElement::getStartColumn()const{return this->start_column;}
int HtmlElement::getTop()const{return this->top;}
int HtmlElement::getWidth()const{return this->width;}
int HtmlElement::getHeight()const{return this->height;}

void HtmlElement::setElementType(std::string elementype){this->element = elementype;}
void HtmlElement::setId(std::string id){this->id = id;}
void HtmlElement::setStartColumn(int s){this->start_column = s;}
void HtmlElement::setTop(int s){this->top = s;}
void HtmlElement::setWidth(int s){this->width = s;}
void HtmlElement::setHeight(int s){this->height = s;}